packages/browser/src/providers/bitski-browser-engine.ts

```diff
@@ -19,43 +21,50 @@ export class BitskiBrowserEngine extends BitskiEngine {
   private clientId: string;
   private sdkVersion: string;
   private headers: object;
+  private signer: BitskiTransactionSigner;
 
   constructor(
     clientId: string,
     tokenProvider: AccessTokenProvider,
     sdkVersion: string,
     network: Network,
-    webBaseUrl?: string,
-    apiBaseUrl?: string,
-    options?: ProviderOptions) {
+    options: ProviderOptions = {},
+  ) {
     super(options);
     this.network = network;
-    this.apiBaseUrl = apiBaseUrl || BITSKI_TRANSACTION_API_BASE_URL;
-    this.tokenProvider = tokenProvider;
     this.clientId = clientId;
-    this.webBaseUrl = webBaseUrl || BITSKI_WEB_BASE_URL;
     this.sdkVersion = sdkVersion;
+    this.apiBaseUrl = options.apiBaseUrl || BITSKI_TRANSACTION_API_BASE_URL;
+    this.webBaseUrl = options.webBaseUrl || BITSKI_WEB_BASE_URL;
+    this.tokenProvider = tokenProvider;
 
     const defaultHeaders = {
       'X-API-KEY': this.clientId,
       'X-CLIENT-ID': this.clientId,
       'X-CLIENT-VERSION': this.sdkVersion,
     };
+
     this.headers = defaultHeaders;
+
     if (options && options.additionalHeaders) {
       this.headers = Object.assign({}, options.additionalHeaders, this.headers);
     }
 
-    this.on('error', (error) => {
```

Comment:
This is no longer necessary. If the user is logged out, the provider should still be allowed to run public requests.